smartsim/entity/application.py

```diff
@@ -170,7 +184,20 @@ def file_parameters(self, value: t.Mapping[str, str]) -> None:
         """Set the file parameters.
 
         :param value: the file parameters
+        :raises TypeError: file_parameters argument is not a mapping of str and str
         """
+        if not (
+            isinstance(value, t.Mapping)
+            and (
+                all(
+                    isinstance(key, str) and isinstance(val, str)
+                    for key, val in value.items()
+                )
+            )
```

Comment:
Style nit if you want to save yourself the indents:
```suggestion
            and all(
                isinstance(key, str) and isinstance(val, str)
                for key, val in value.items()
            )
```